src/Lib.hs

```diff
@@ -43,76 +48,183 @@ startPetSelection user goldRemaining = do
 
 startBattle :: User -> IO ()
 startBattle user = do
-    putStrBar
-    putStrLn $ show (userRoster user)
-    putStrLn "\nBATTLE"
-    putStrLn "BATTLE"
-    putStrLn "BATTLE"
-    -- TODO BATTLE
-
-
-putStrBar :: IO ()
-putStrBar = do
-    putStrLn "\n==================="
+    printBar
+    putStrLn "YOUR TEAM"
+    putStrLn $ display (userRoster user)
+    
+    -- Create opponent team
+    petOpponent1 <- getPet allPets
+    petOpponent2 <- getPet allPets
+    petOpponent3 <- getPet allPets
+    let 
+      opponentRoster = Roster
+        { rosterPet1 = Just petOpponent1
+        , rosterPet2 = Just petOpponent2
+        , rosterPet3 = Just petOpponent3
+        , rosterPet4 = Nothing
+        , rosterPet5 = Nothing
+        }
+    let
+      userOpponent = User
+        { userName = "OPPONENT"
+        , userRoster = opponentRoster
+        , userItemList = itemListEmpty
+        }
+    putStrLn "\nOPPONENT TEAM"
+    putStrLn $ display (opponentRoster)
+
+    -- BATTLE
+    battleRoster user userOpponent
+
+
+battleRoster :: User -> User -> IO ()
+battleRoster user1 user2 = do
+  -- get pets
+  printBar
+  putStrLn "NEW CHALLENGER:"
+  let r1pet = getRosterFirstWhere (userRoster user1) healthPositive
+  let r2pet = getRosterFirstWhere (userRoster user2) healthPositive
+  let user1Pet = fromJust r1pet
+  let user2Pet = fromJust r2pet
+  printPetBattle user1Pet user2Pet
+
+  -- battle pets
+  let (user1Pet', user2Pet') = battlePets user1Pet user2Pet
+  putStrLn "\nResult:"
+  printPetBattle user1Pet' user2Pet'
+
+  -- create updated rosters
+  let r1' = replacePet user1Pet user1Pet' (userRoster user1)
+  let r2' = replacePet user2Pet user2Pet' (userRoster user2)
+  
+  -- create updated users
+  let user1' = user1 { userRoster = r1' }
+  let user2' = user2 { userRoster = r2' }
+  
+  threadDelay 3000000 --sleep for a million microseconds, or one second
+
+  -- detect if either player is out of pets
+  -- TODO how to handle ties?
```

Comment:
Maybe consider returning some data structure which represents the result. From the player perspective there are 3 such results.